pandas/sparse/array.py

```diff
@@ -322,31 +332,51 @@ def take(self, indices, axis=0):
             # return scalar
             return self[indices]
 
-        indices = np.atleast_1d(np.asarray(indices, dtype=int))
-
-        # allow -1 to indicate missing values
+        indices = com._ensure_platform_int(indices)
```

Comment:
Ah, this can't be simple because ``IndexError`` is handled here rather than numpy. Can I leave it ATM?